litestar/_kwargs/extractors.py

```diff
@@ -348,7 +348,15 @@ async def _extract_multipart(
 
     if field_definition.is_non_string_sequence:
         values = list(form_values.values())
-        if field_definition.has_inner_subclass_of(UploadFile) and isinstance(values[0], list):
+        if (
+            values
+            and isinstance(values[0], list)
+            and (
+                field_definition.has_inner_subclass_of(UploadFile)
+                or field_definition.is_optional
+                and any(fd.has_inner_subclass_of(UploadFile) for fd in field_definition.inner_types)
```

Comment:
```suggestion
                field_definition.has_inner_subclass_of(UploadFile)
                or (field_definition.is_optional
                and any(fd.has_inner_subclass_of(UploadFile) for fd in field_definition.inner_types))
```

The optional check and the check on the inner types should be grouped together right?

Reply:
I don't think so because `and` already has higher [operator precedence][1] than `or` so that grouping is implicit.

E.g., we have `a or b and c`, if `a` is truthy, the conditional short-circuits and returns the value for `a`.

```python-console
>>> True or 0 and "truthy"
True
```

If `a` is falsy, then the rhs starts to get evaluated left to right:

```python-console
>>> False or 0 and "truthy"
0
>>> False or 1 and "truthy"
'truthy'
```

So for the suggested change to be required, we'd need `True or 0 and "truthy" == "truthy"` to hold, but it doesn't.

[1]: https://docs.python.org/3/reference/expressions.html#operator-precedence